src/Field.php

```diff
@@ -210,7 +210,7 @@ public function __construct(array $defaults = [])
     {
         foreach ($defaults as $key => $val) {
             if (is_array($val)) {
-                $this->{$key} = array_merge(is_array($this->{$key} ?? null) ? $this->{$key} : [], $val);
+                $this->{$key} = array_merge_recursive(is_array($this->{$key} ?? null) ? $this->{$key} : [], $val);
```

Comment:
this is in a loop already, so it is quite dangerous to merge recursively, which can/would result with mismatched values

Reply:
In a loop yes, but defaults can be arrays too and these arrays should be merged recursively otherwise it's not possible to extend them - only fully overwrite.

I agree about adding BC break and removing Bug.

Reply:
yes, but merging elements can result in mismatch, what is the exact usecase we need to deep merge?